zokrates_core/src/semantics.rs

```diff
@@ -2155,6 +2053,153 @@ impl<'ast, T: Field> Checker<'ast, T> {
         }
     }
 
+    fn check_function_call_expression(
+        &mut self,
+        function_id: ExpressionNode<'ast>,
+        generics: Option<Vec<Option<ExpressionNode<'ast>>>>,
+        arguments: Vec<ExpressionNode<'ast>>,
+        assignee_type: Option<Type<'ast, T>>,
```

Comment:
How about `expected_return_type`? We use this function outside of the assignee context also.